.github/workflows/ccpp.yml

```diff
@@ -0,0 +1,106 @@
+name: DOLFINX CI tests
+
+on:
+  push:
+    branches:
+      - "*"
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    container: fenicsproject/test-env
+
+    env:
+      CC: clang-10
+      CXX: clang++-10
+      DOLFINX_JIT_CFLAGS: -g0 -O0
+      MPLBACKEND: agg
+      PETSC_ARCH: linux-gnu-${{ matrix.petsc_arch }}-${{ matrix.petsc_int_type }}
+      OMPI_ALLOW_RUN_AS_ROOT: 1
+      OMPI_ALLOW_RUN_AS_ROOT_CONFIRM: 1
+      OMPI_MCA_rmaps_base_oversubscribe: 1
+      OMPI_MCA_plm: isolated
+      OMPI_MCA_btl_vader_single_copy_mechanism: none
+      OMPI_MCA_mpi_yield_when_idle: 1
+      OMPI_MCA_hwloc_base_binding_policy: none
+
+    strategy:
+      matrix:
+        petsc_arch: [real, complex]
+        petsc_int_type: [32]
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install FEniCS Python components
+        run: |
+          python3 -m pip install git+https://github.com/FEniCS/fiat.git
+          python3 -m pip install git+https://github.com/FEniCS/ufl.git
+          python3 -m pip install git+https://github.com/FEniCS/ffcx.git
+      - name: Install pybind11 (development branch)
+        run: |
+          git clone --depth=1 -b master https://github.com/pybind/pybind11.git
+          cd pybind11
+          mkdir build
+          cd build
+          cmake -DPYBIND11_TEST=False ../
+          make install
+
+      - name: Flake8 checks
+        run: |
+          python3 -m flake8 python/dolfinx
+          python3 -m flake8 python/dolfinx_utils
+          python3 -m flake8 python/demo
+          python3 -m flake8 python/test
+      - name: Build C++ interface documentation
+        run: cd cpp/doc && doxygen
+
+      - name: Configure C++
+        run: cmake -G Ninja -DCMAKE_BUILD_TYPE=Developer -B build -S cpp/
+      - name: Build and install C++ library
+        run: |
+          cmake --build build
+          cmake --install build
+
+      - name: Build and run C++ unit tests (serial and MPI)
+        run: |
+          cmake --build ./build -t unittests
```

Comment:
This line (and the similar one in demos) would change in #897. Let me know if you are willing to merge this before #897, and I will rebase accordingly.

Reply:
I can merge this first.

Reply:
OK, I will update #897 as soon as this is merged.